.github/workflows/changelog_to_tag.yml

```diff
@@ -19,30 +19,44 @@ jobs:
         id: tag
         run: |
           set -euxo pipefail
-          pat='\[[0-9]*\.[0-9]*\.[0-9]*\] - [0-9\-]*'
           print=false
-          cat CHANGELOG.md | while read -r line; do
-              if [[ "$line" =~ $pat ]] && [[ "$print" == false ]]; then
-                  echo "$line"
-                  print=true
-              elif [[ "$line" =~ $pat ]] && [[ "$print" == true ]]; then
-                  break
-              elif [[ "$print" == true ]]; then
+          while read -r line; do
+              if [[ "$line" =~ ^\[([0-9]+\.[0-9]+\.[0-9]+)\]\ -\ [0-9-]+ ]]; then
+                  if [ "$print" = false ]; then
+                      _tagname="${BASH_REMATCH[1]}"
+                      echo "$line"
+                      print=true
+                  else
+                      break
+                  fi
+              elif [ "$print" = true ]; then
                   echo "$line"
               fi
-          done > ./.tagmsg.txt
-          _tagname=$( grep -m 1 "[0-9]*\.[0-9]*\.[0-9]*" CHANGELOG.md | \
-                      sed -e "s/^.*\[\([0-9]*\.[0-9]*\.[0-9]*\)\].*/\1/" )
+          done < CHANGELOG.md > ./.tagmsg.txt
           git fetch --all --tags
           for t in $( git tag -l ); do
-              if [[ $t == "$_tagname" ]]; then
-                  echo INFO: tag $t already exists
+              if [ "$t" = "$_tagname" ]; then
+                  echo INFO: tag "$t" already exists
                   exit 1
               fi
           done
-          # Get the main branch name, "master" or "main".
-          _branch=$( git branch -r | grep -o 'origin/HEAD -> origin/.*$' | \
-                     awk -F'/' '{print $3}' )
+          # Get name of the branch that the change was pushed to
+          _branch="${GITHUB_REF_NAME:-}"
+          if [ "$_branch" = master ] || [ "$_branch" = main ]; then
+              echo Using branch name ["$_branch"] as push branch
+          else
+              echo WARNING: GITHUB_REF_NAME ["$_branch"] is not main or master
```

Comment:
I wonder if $GITHUB_REF_NAME is not `main` and `master`, it matches the one retrieved by the next line 49? Maybe, the line 49 is not needed? (I'd think issuing the warning at line 48 is a good idea.)

Reply:
> I wonder if $GITHUB_REF_NAME is not `main` and `master`, it matches the one retrieved by the next line 49?

I don't know.  My concern is that in the future `GITHUB_REF_NAME` may contain something that isn't a simple branch name e.g. maybe it would be `origin/main` or `refs/main`.  Or, maybe github decides to drop it altogether, in which case `$_branch` will be `""`.  In any of those cases, we would need to figure out the branch name some other way.

> Maybe, the line 49 is not needed? (I'd think issuing the warning at line 48 is a good idea.)

I hope line 49 is not needed.  I hope that we can trust that `GITHUB_REF_NAME` is always, now and in the future, the name of the push branch.

So another way to do this would be 
```
               echo ERROR: GITHUB_REF_NAME ["$_branch"] is not main or master
               exit 1
```
and that's that - either we use `GITHUB_REF_NAME` or exit with an error.  That would make the code much simpler.

Reply:
I see. Indeed it's hard to predict if `GITHUB_REF_NAME` is guaranteed to be the same. I prefer your simpler proposal above, but the redundancy might pay off some day... Thanks for the explanation.